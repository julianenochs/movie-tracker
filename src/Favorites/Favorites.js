import React, { Component } from 'react';
import { connect } from 'react-redux';
import MovieCard from '../components/MovieCard/MovieCard';
import './Favorites.scss'
class Favorites extends Component {
  returnFavorites = () => {
    return this.props.favorites.map((fav, i) => {
      return (
        <MovieCard 
          title={fav.title}
          poster={fav.poster_path}
          overview={fav.overview}
          movieID={fav.movie_id}
          key={i}
        />
      );
    });
  }

  render() {
    return (
      <section className='favorites__container'>
        {this.returnFavorites()}
      </section>
    );
  }
}
const mapStateToProps = state => ({
  favorites: state.favorites,
  movies: state.movies,
});

export default connect(
  mapStateToProps,
  null
)(Favorites);
