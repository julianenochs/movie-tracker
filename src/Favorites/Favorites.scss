.favorites__container {
  align-items: center;
  background-color: rgba(0, 31, 46, 0.78);
  color: white;
  display: flex;
  flex-direction: row;
  margin: 3px;
  margin-block-start: 5px;
  height: 60vh;
  overflow: scroll;
  overflow-y: hidden;
  padding: 15px;
  width: 85%;
}