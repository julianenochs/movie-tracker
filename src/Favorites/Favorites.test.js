import React from 'react';
import { shallow } from 'enzyme';
import { Favorites, mapStateToProps, mapDispatchToProps } from './Favorites';

describe('Favorites', () => {
  let wrapper;
  const mockMovies = [
    {
      popularity: 471.315,
      vote_count: 4211,
      video: false,
      poster_path: '/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg',
      id: 475557,
      adult: false,
      backdrop_path: '/n6bUvigpRFqSwmPp1m2YADdbRBc.jpg',
      original_language: 'en',
      original_title: 'Joker',
      title: 'Joker',
      vote_average: 8.6,
      overview:
        'During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.',
      release_date: '2019-10-04',
      isFavorite: false
    }
  ];

  beforeEach(() => {
    wrapper = shallow(<Favorites favorites={[]} movies={mockMovies} />);
  });

  it('should match the Favorites Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
