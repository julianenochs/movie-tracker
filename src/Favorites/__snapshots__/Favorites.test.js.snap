// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match the Favorites Snapshot 1`] = `ShallowWrapper {}`;
